clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClientTest.java

```diff
@@ -268,26 +270,39 @@ public void testMetadataFailurePropagated() {
     public void testFutureCompletionOutsidePoll() throws Exception {
         // Tests the scenario in which the request that is being awaited in one thread
         // is received and completed in another thread.
+        
+        final CountDownLatch t1TheardCountDownLatch = new CountDownLatch(1);
+        final CountDownLatch t2ThreadCountDownLatch = new CountDownLatch(2);
 
         final RequestFuture<ClientResponse> future = consumerClient.send(node, heartbeat());
         consumerClient.pollNoWakeup(); // dequeue and send the request
 
         client.enableBlockingUntilWakeup(2);
-        Thread t1 = new Thread(() -> consumerClient.pollNoWakeup());
+        Thread t1 = new Thread(() -> assertDoesNotThrow(() -> {
+            t1TheardCountDownLatch.countDown();
+            consumerClient.pollNoWakeup();
+            t2ThreadCountDownLatch.countDown();
+        }));
+        
         t1.start();
 
-        // Sleep a little so that t1 is blocking in poll
-        Thread.sleep(50);
-
-        Thread t2 = new Thread(() -> consumerClient.poll(future));
+        Thread t2 = new Thread(() -> assertDoesNotThrow(() -> {
+            t2ThreadCountDownLatch.await();
+            consumerClient.poll(future);
+        }));
         t2.start();
-
-        // Sleep a little so that t2 is awaiting the network client lock
-        Thread.sleep(50);
-
+        
         // Simulate a network response and return from the poll in t1
         client.respond(heartbeatResponse(Errors.NONE));
+        // Wait for t1 to block in poll
+        t1TheardCountDownLatch.await();
+        // Sleep a little so that t1 is blocking in poll
+        Thread.sleep(50);
```

Comment:
why not calling `t2ThreadCountDownLatch.await();` instead?

Reply:
when t2 thread start, the `t2ThreadCountDownLatch.await()` is calling, I think we should not call `t2ThreadCountDownLatch.await()` in there again.

Reply:
> when t2 thread start, the t2ThreadCountDownLatch.await() is calling, I think we should not call t2ThreadCountDownLatch.await() in there again.

why we can't call `t2ThreadCountDownLatch` again? `Thread.sleep(50);` can't ensure t1 is in blocking, right?

Reply:
But `t2ThreadCountDownLatch` init value is 2, so we can ensure t2 thread will wakeup after main thread `client.wakeup()` and t1 thread is `consumerClient.pollNoWakeup()` 
I think the main flow is 
1. `main` thread is await `t1` thread to wake up
2. t1 thread `pollNoWakeup` and `main` thread call `client.wakeup()`
3. `t1` thread wake up `t2` thread because `t2ThreadCountDownLatch` countdown in `main` thread and `t1` thread